src/Services/QueuedJobService.php

```diff
@@ -1533,6 +1579,271 @@ private function addJobHandlersToLogger(LoggerInterface $logger, QueuedJob $job,
         }
     }
 
+    /**
+     * Find any eligible jobs that qualify for an automated job retry and schedule their retry
+     *
+     * @throws ValidationException
+     */
+    protected function retryEligibleJobs(int $queueType): void
+    {
+        // Find any broken jobs that qualify for an automated job retry
+        $jobRetrySentinel = (int) static::config()->get('job_retry_sentinel');
+        $jobRetryLimit = (int) static::config()->get('job_retry_limit');
+        $jobRetryStatus = (array) static::config()->get('job_retry_status');
+
+        $jobStatusMap = [];
+        $jobRetrySentinel = max(0, $jobRetrySentinel);
+
+        // Job retries feature is disabled
+        if ($jobRetryLimit <= 0) {
+            return;
+        }
+
+        $validJobStatuses = QueuedJobDescriptor::singleton()->getJobStatusValues();
+
+        foreach ($jobRetryStatus as $retryStatus => $targetStatus) {
+            // Skip any disabled status conditions
+            if (!$targetStatus) {
+                continue;
+            }
+
+            // Skip any invalid status conditions
+            if (!in_array($targetStatus, $validJobStatuses)) {
+                continue;
+            }
+
+            $jobStatusMap[$retryStatus] = $targetStatus;
+        }
+
+        $jobStatusConditions = array_keys($jobStatusMap);
+
+        // We don't have any job status transformation configured
+        if (count($jobStatusConditions) === 0) {
+            return;
+        }
+
+        $now = DBDatetime::now();
+        $lastEditedSentinel = $now
+            // This is important to set otherwise it will break unit tests due to how the date mock works
+            ->setImmutable(true)
+            ->modify(sprintf('-%d minutes', $jobRetrySentinel))
+            ->Rfc2822();
+
+        $autoRetryJobs = QueuedJobDescriptor::get()
+            ->filter([
+                // Automatic retry is applied to only configured jobs statuses
+                'JobStatus' => $jobStatusConditions,
+                // Make sure to avoid retrying jobs that are being processed very recently to avoid edge cases
```

Comment:
```suggestion
                // Make sure to avoid retrying jobs that have been processed very recently to avoid edge cases
```
This is a nitpick - won't delay merging for this on its own